services/src/datasets/external/sentinel_s2_l2a_cogs.rs

```diff
@@ -0,0 +1,640 @@
+                        description: "".to_owned(),
+                        tags: vec![],
+                        source_operator: "GdalSource".to_owned(),
+                        result_descriptor: RasterResultDescriptor {
+                            data_type: band.data_type,
+                            spatial_reference: SpatialReference::new(
+                                SpatialReferenceAuthority::Epsg,
+                                zone.epsg,
+                            )
+                            .into(),
+                            measurement: Measurement::Unitless,
+                            no_data_value: band.no_data_value,
+                        }
+                        .into(),
+                    };
+
+                    let dataset = SentinelDataset {
+                        zone: zone.clone(),
+                        band: band.clone(),
+                        listing,
+                    };
+
+                    (dataset_id, dataset)
+                })
+            })
+            .collect()
+    }
+}
+
+#[async_trait]
+impl DatasetProvider for SentinelS2L2aCogsDataProvider {
+    async fn list(&self, _options: Validated<DatasetListOptions>) -> Result<Vec<DatasetListing>> {
+        // TODO: options
+        Ok(self.datasets.values().map(|d| d.listing.clone()).collect())
+    }
+
+    async fn load(
+        &self,
+        _dataset: &geoengine_datatypes::dataset::DatasetId,
+    ) -> crate::error::Result<crate::datasets::storage::Dataset> {
+        Err(error::Error::NotYetImplemented)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct SentinelS2L2aCogsMetaData {
+    api_url: String,
+    zone: Zone,
+    band: Band,
+}
+
+impl SentinelS2L2aCogsMetaData {
+    async fn create_loading_info(&self, query: QueryRectangle) -> Result<GdalLoadingInfo> {
+        // for reference: https://stacspec.org/STAC-ext-api.html#operation/getSearchSTAC
+
+        let features = self.load_all_features(&self.request_params(query)?).await?;
+        debug!("number of features returned by STAC: {}", features.len());
+        let mut features: Vec<StacFeature> = features
+            .into_iter()
+            .filter(|f| {
+                f.properties
+                    .proj_epsg
+                    .map_or(false, |epsg| epsg == self.zone.epsg)
+            })
+            .collect();
+
+        features.sort_by_key(|a| a.properties.datetime);
+
+        let mut parts = vec![];
+        let num_features = features.len();
+        debug!("number of features in current zone: {}", num_features);
+        for i in 0..num_features {
+            let feature = &features[i];
+
+            let start = feature.properties.datetime;
+            // feature is valid until next feature starts
+            let end = if i < num_features - 1 {
+                features[i + 1].properties.datetime
+            } else {
+                start + Duration::hours(12) // TODO: determine correct validity for last tile
```

Comment:
maybe just add 1 minute. this should be fine..

Reply:
are you sure? is the satellite also recording while it's on the night side?

Reply:
as above

Reply:
if I only shift by one minute there may be no tile at all because the latest tile is from previous day

Reply:
yes but i think this is ok. ypu need to aggregate anyway

Reply:
89397a1bc02a646b760c2b2655eec984131b749b